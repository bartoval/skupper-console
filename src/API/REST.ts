import { HttpStatusErrors } from '@pages/shared/Errors/errors.constants';

import { fetchWithTimeout, handleStatusError } from './axiosMiddleware';
import { getFlowsTopology, getServices, getSites } from './controllers';
import {
    DATA_URL,
    FLOWS_CONNECTORS,
    FLOWS_LINKS,
    FLOWS_LISTENERS,
    SITES_PATH,
    FLOWS_VAN_ADDRESSES,
    FLOWPAIRS,
    getFlowsPairsByVanAddressIdPATH,
    getProcessesBySitePATH,
    getProcessesByVanAddressIdPATH,
    getConnectorByProcessIdPATH,
    getFlowsByProcessIdPATH,
    getSitePATH,
    getRoutersBySitePATH,
    getLinksBySitePATH,
    getHostsBySitePATH,
    FLOW_AGGREGATES_PROCESSES,
    ROUTERS_PATH,
    PROCESSES_PATH,
    FLOW_AGGREGATES_SITES,
    HOSTS_PATH,
} from './REST.constant';
import {
    ServiceResponse,
    SiteDataResponse,
    FlowsVanAddressesResponse,
    FlowsTopologyResponse,
    DeviceResponse,
    ProcessResponse,
    FlowPairResponse,
    HTTPError,
    FlowResponse,
    SiteResponse,
    LinkResponse,
    RouterResponse,
    HostResponse,
    FlowAggregatesMapResponse,
    FlowAggregatesResponse,
} from './REST.interfaces';

export const RESTApi = {
    fetchDATASites: async (): Promise<SiteDataResponse[]> => {
        const { data } = await fetchWithTimeout(DATA_URL);

        return getSites(data);
    },

    // SITES APIs
    fetchSites: async (): Promise<SiteResponse[]> => {
        const { data } = await fetchWithTimeout(SITES_PATH);

        return data;
    },
    fetchSite: async (id: string): Promise<SiteResponse> => {
        const { data } = await fetchWithTimeout(getSitePATH(id));

        return data;
    },
    fetchProcessesBySite: async (id: string): Promise<ProcessResponse[]> => {
        const { data } = await fetchWithTimeout(getProcessesBySitePATH(id));

        return data;
    },
    fetchRoutersBySite: async (id: string): Promise<RouterResponse[]> => {
        const { data } = await fetchWithTimeout(getRoutersBySitePATH(id));

        return data;
    },
    fetchLinksBySite: async (id: string): Promise<LinkResponse[]> => {
        const { data } = await fetchWithTimeout(getLinksBySitePATH(id));

        return data;
    },
    fetchHostsBySite: async (id: string): Promise<HostResponse[]> => {
        const { data } = await fetchWithTimeout(getHostsBySitePATH(id));

        return data;
    },

    // ROUTER APIs
    fetchRouters: async (): Promise<RouterResponse[]> => {
        const { data } = await fetchWithTimeout(`${ROUTERS_PATH}`);

        return data;
    },
    fetchRouter: async (id: string): Promise<RouterResponse> => {
        const { data } = await fetchWithTimeout(`${ROUTERS_PATH}/${id}`);

        return data;
    },

    // PROCESS APIs
    fetchProcesses: async (): Promise<ProcessResponse[]> => {
        const { data } = await fetchWithTimeout(`${PROCESSES_PATH}`);

        return data;
    },

    fetchProcess: async (id: string): Promise<ProcessResponse> => {
        const { data } = await fetchWithTimeout(`${PROCESSES_PATH}/${id}`);

        return data;
    },

    // HOST APIs
    fetchHost: async (): Promise<HostResponse[]> => {
        const { data } = await fetchWithTimeout(`${HOSTS_PATH}`);

        return data;
    },

    // SERVICES APIs
    fetchServices: async (): Promise<ServiceResponse[]> => {
        const { data } = await fetchWithTimeout(DATA_URL);

        return getServices(data);
    },

    // FLOWS APIs
    fetchVanAddresses: async (): Promise<FlowsVanAddressesResponse[]> => {
        const { data } = await fetchWithTimeout(FLOWS_VAN_ADDRESSES);

        return data;
    },
    fetchFlowsSites: async (): Promise<SiteResponse[]> => {
        const { data } = await fetchWithTimeout(`${SITES_PATH}`);

        return data;
    },
    fetchFlowsSite: async (id: string): Promise<SiteResponse> => {
        const { data } = await fetchWithTimeout(`${SITES_PATH}/${id}`);

        return data;
    },
    fetchFlowsByProcessesId: async (id: string): Promise<FlowResponse[]> => {
        const { data } = await fetchWithTimeout(getFlowsByProcessIdPATH(id));

        return data;
    },

    fetchFlowConnectorByProcessId: async (id: string): Promise<DeviceResponse> => {
        const { data } = await fetchWithTimeout(getConnectorByProcessIdPATH(id));

        return data;
    },
    fetchFlowsLinks: async (): Promise<DeviceResponse[]> => {
        const { data } = await fetchWithTimeout(`${FLOWS_LINKS}`);

        return data;
    },
    fetchFlowsLink: async (id: string): Promise<DeviceResponse> => {
        const { data } = await fetchWithTimeout(`${FLOWS_LINKS}/${id}`);

        return data;
    },
    fetchFlowsConnectors: async (): Promise<DeviceResponse[]> => {
        const { data } = await fetchWithTimeout(`${FLOWS_CONNECTORS}`);

        return data;
    },
    fetchFlowsConnector: async (id: string): Promise<DeviceResponse | null> => {
        try {
            const { data } = await fetchWithTimeout(`${FLOWS_CONNECTORS}/${id}`);

            return data;
        } catch (e) {
            const error = e as HTTPError;

            if (error.httpStatus === HttpStatusErrors.NotFound) {
                return null;
            }

            return handleStatusError(error);
        }
    },
    fetchFlowsListeners: async (): Promise<DeviceResponse[]> => {
        const { data } = await fetchWithTimeout(`${FLOWS_LISTENERS}`);

        return data;
    },
    fetchFlowsListener: async (id: string): Promise<DeviceResponse> => {
        const { data } = await fetchWithTimeout(`${FLOWS_LISTENERS}/${id}`);

        return data;
    },
    fetchFlowsPairsByVanAddr: async (id: string): Promise<FlowPairResponse[]> => {
        const { data } = await fetchWithTimeout(getFlowsPairsByVanAddressIdPATH(id));

        return data;
    },
    fetchProcessesByVanAddr: async (id: string): Promise<ProcessResponse[]> => {
        const { data } = await fetchWithTimeout(getProcessesByVanAddressIdPATH(id));

        return data;
    },
    fetchFlowPair: async (id: string): Promise<FlowPairResponse> => {
        const { data } = await fetchWithTimeout(`${FLOWPAIRS}/${id}`);

        return data;
    },
    fetchFlowsTopology: async (): Promise<FlowsTopologyResponse> => {
        const { data: sites } = await fetchWithTimeout(`${SITES_PATH}`);
        const { data: routers } = await fetchWithTimeout(`${ROUTERS_PATH}`);
        const { data: links } = await fetchWithTimeout(`${FLOWS_LINKS}`);

        const sitesMap = (sites as SiteResponse[]).reduce((acc, site) => {
            acc[site.identity] = site;

            return acc;
        }, {} as Record<string, SiteResponse>);

        const routersExtended = (routers as RouterResponse[]).map((router) => ({
            ...router,
            siteName: sitesMap[router.parent].name,
        }));

        return getFlowsTopology(routersExtended, links);
    },

    // FLOWAGGREGATES APIs
    fetchFlowAggregatesSites: async (): Promise<FlowAggregatesMapResponse[]> => {
        const { data } = await fetchWithTimeout(FLOW_AGGREGATES_SITES);

        return data;
    },

    fetchFlowAggregatesSite: async (id: string): Promise<FlowAggregatesResponse> => {
        const { data } = await fetchWithTimeout(`${FLOW_AGGREGATES_SITES}/${id}`);

        return data;
    },

    fetchFlowAggregatesProcesses: async (): Promise<FlowAggregatesMapResponse[]> => {
        const { data } = await fetchWithTimeout(FLOW_AGGREGATES_PROCESSES);

        return data;
    },

    fetchFlowAggregatesProcess: async (id: string): Promise<FlowAggregatesResponse> => {
        const { data } = await fetchWithTimeout(`${FLOW_AGGREGATES_PROCESSES}/${id}`);

        return data;
    },
};
