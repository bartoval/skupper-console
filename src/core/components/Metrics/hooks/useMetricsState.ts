import { useCallback, useState, startTransition } from 'react';

import { useMetricSessionHandlers } from './useMetricsSessionHandler';
import { QueryMetricsParams } from '../../../../types/Metrics.interfaces';

interface UseMetricsProps {
  sessionKey?: string;
  defaultMetricFilterValues: QueryMetricsParams;
}

export const useMetricsState = ({ sessionKey, defaultMetricFilterValues }: UseMetricsProps) => {
  const { selectedFilters: storedFilters, setSelectedFilters: setStoredFilters } = useMetricSessionHandlers(
    sessionKey,
    defaultMetricFilterValues
  );

  const [selectedFilters, setSelectedFilters] = useState<QueryMetricsParams>(storedFilters);

  const handleFilterChange = useCallback(
    (updatedFilters: QueryMetricsParams) => {
      startTransition(() => {
        // let to override undefined values with default values if they exists
        const filters = { ...defaultMetricFilterValues, ...JSON.parse(JSON.stringify({ ...updatedFilters })) };

        setStoredFilters(filters);
        setSelectedFilters(filters);
      });
    },
    [defaultMetricFilterValues, setStoredFilters]
  );

  return {
    selectedFilters,
    handleFilterChange
  };
};
