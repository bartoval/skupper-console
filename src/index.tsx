import React from 'react';

import { createRoot } from 'react-dom/client';
import { QueryClient, QueryClientProvider } from 'react-query';
import { HashRouter } from 'react-router-dom';

import App from '@core/components/App';

import { loadMockServer } from '../public/mocks/server.js';

const queryClient = new QueryClient();
const rootElement = document.getElementById('app') as HTMLDivElement;
const root = createRoot(rootElement);

root.render(
    <QueryClientProvider client={queryClient}>
        <HashRouter>
            <App />
        </HashRouter>
    </QueryClientProvider>,
);

loadMockServer();
