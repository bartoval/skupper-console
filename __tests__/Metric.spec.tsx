import { Suspense } from 'react';

import { act, render, renderHook } from '@testing-library/react';

import { Protocols } from '../src/API/REST.enum';
import LoadingPage from '../src/core/components/SkLoading';
import { Providers } from '../src/providers';
import Metrics, { MetricsProps } from '../src/pages/shared/Metrics';
import { useMetricsState } from '../src/pages/shared/Metrics/hooks/useMetricsState';
import { configDefaultFilters } from '../src/pages/shared/Metrics/Metrics.constants';

describe('useMetrics', () => {
  const initialProps = {
    defaultMetricFilterValues: { protocol: Protocols.Http },
    defaultOpenSections: { byterate: true },
    onGetMetricFiltersConfig: jest.fn(),
    onGetExpandedSectionsConfig: jest.fn()
  };

  it('initializes with the correct state from props', () => {
    const { result } = renderHook(() => useMetricsState(initialProps));
    expect(result.current.queryParams).toEqual({ protocol: Protocols.Http });
    expect(result.current.shouldUpdateData).toBe(0);
  });

  it('updates shouldUpdateData correctly when triggerMetricUpdate is called', () => {
    const { result } = renderHook(() => useMetricsState(initialProps));
    act(() => {
      result.current.triggerMetricUpdate();
    });
    expect(result.current.shouldUpdateData).not.toBe(0);
  });

  it('updates queryParams and calls onGetMetricFiltersConfig when handleFilterChange is invoked', () => {
    const { result } = renderHook(() => useMetricsState(initialProps));
    const newFilters = { protocol: Protocols.Tcp };
    act(() => {
      result.current.handleFilterChange(newFilters);
    });
    expect(result.current.queryParams).toEqual(newFilters);
    expect(initialProps.onGetMetricFiltersConfig).toHaveBeenCalledWith(newFilters);
  });

  it('updates expanded sections and calls onGetExpandedSectionsConfig when handleSectionToggle is invoked', () => {
    const { result } = renderHook(() => useMetricsState(initialProps));
    const newSections = { traffic: false };
    act(() => {
      result.current.handleSectionToggle(newSections);
    });
    expect(initialProps.onGetExpandedSectionsConfig).toHaveBeenCalledWith({
      ...initialProps.defaultOpenSections,
      ...newSections
    });
  });
});

describe('Metrics Component', () => {
  const setup = (overrides?: Partial<MetricsProps>) => ({
    configFilters: configDefaultFilters,
    defaultMetricFilterValues: {},
    defaultOpenSections: {},
    sourceSites: [],
    destSites: [],
    sourceProcesses: [],
    destProcesses: [],
    availableProtocols: [Protocols.Http, Protocols.Tcp, Protocols.Http2],
    onGetMetricFiltersConfig: jest.fn(),
    onGetExpandedSectionsConfig: jest.fn(),
    ...overrides
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Providers>
        <Suspense fallback={<LoadingPage />}>
          <Metrics {...setup()} />
        </Suspense>
      </Providers>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
