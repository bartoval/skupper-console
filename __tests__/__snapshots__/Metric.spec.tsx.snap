// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics Component renders correctly 1`] = `
<DocumentFragment>
  <div
    style="position: sticky; top: 0px; z-index: 1; background-color: rgb(255, 255, 255);"
  >
    <div
      class="pf-v6-c-toolbar"
      data-ouia-component-id="OUIA-Generated-Toolbar-1"
      data-ouia-component-type="PF6/Toolbar"
      data-ouia-safe="true"
      id="pf-random-id-0"
    >
      <div
        class="pf-v6-c-toolbar__content"
      >
        <div
          class="pf-v6-c-toolbar__content-section"
        >
          <div
            class="pf-v6-c-toolbar__group"
          >
            <div
              class="pf-v6-c-toolbar__item"
            >
              <button
                aria-expanded="false"
                class="pf-v6-c-menu-toggle"
                data-ouia-component-id="OUIA-Generated-MenuToggle-1"
                data-ouia-component-type="PF6/MenuToggle"
                data-ouia-safe="true"
                type="button"
              >
                <span
                  class="pf-v6-c-menu-toggle__icon"
                >
                  <div
                    style="display: contents;"
                  >
                    <span
                      class="sk-resource-icon"
                      role="site-resource-icon"
                      style="background: rgb(99, 153, 61);"
                    >
                      S
                    </span>
                  </div>
                </span>
                <span
                  class="pf-v6-c-menu-toggle__text"
                >
                  All source sites
                </span>
                <span
                  class="pf-v6-c-menu-toggle__controls"
                >
                  <span
                    class="pf-v6-c-menu-toggle__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="pf-v6-c-toolbar__item"
            >
              <button
                aria-expanded="false"
                class="pf-v6-c-menu-toggle"
                data-ouia-component-id="OUIA-Generated-MenuToggle-2"
                data-ouia-component-type="PF6/MenuToggle"
                data-ouia-safe="true"
                type="button"
              >
                <span
                  class="pf-v6-c-menu-toggle__icon"
                >
                  <div
                    style="display: contents;"
                  >
                    <span
                      class="sk-resource-icon"
                      role="process-resource-icon"
                      style="background: rgb(112, 112, 112);"
                    >
                      P
                    </span>
                  </div>
                </span>
                <span
                  class="pf-v6-c-menu-toggle__text"
                >
                  All source processes
                </span>
                <span
                  class="pf-v6-c-menu-toggle__controls"
                >
                  <span
                    class="pf-v6-c-menu-toggle__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="pf-v6-c-toolbar__item"
            >
              <button
                aria-expanded="false"
                class="pf-v6-c-menu-toggle"
                data-ouia-component-id="OUIA-Generated-MenuToggle-3"
                data-ouia-component-type="PF6/MenuToggle"
                data-ouia-safe="true"
                type="button"
              >
                <span
                  class="pf-v6-c-menu-toggle__icon"
                >
                  <div
                    style="display: contents;"
                  >
                    <span
                      class="sk-resource-icon"
                      role="site-resource-icon"
                      style="background: rgb(99, 153, 61);"
                    >
                      S
                    </span>
                  </div>
                </span>
                <span
                  class="pf-v6-c-menu-toggle__text"
                >
                  All connected sites
                </span>
                <span
                  class="pf-v6-c-menu-toggle__controls"
                >
                  <span
                    class="pf-v6-c-menu-toggle__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="pf-v6-c-toolbar__item"
            >
              <button
                aria-expanded="false"
                class="pf-v6-c-menu-toggle"
                data-ouia-component-id="OUIA-Generated-MenuToggle-4"
                data-ouia-component-type="PF6/MenuToggle"
                data-ouia-safe="true"
                type="button"
              >
                <span
                  class="pf-v6-c-menu-toggle__icon"
                >
                  <div
                    style="display: contents;"
                  >
                    <span
                      class="sk-resource-icon"
                      role="process-resource-icon"
                      style="background: rgb(112, 112, 112);"
                    >
                      P
                    </span>
                  </div>
                </span>
                <span
                  class="pf-v6-c-menu-toggle__text"
                >
                  All connected processes
                </span>
                <span
                  class="pf-v6-c-menu-toggle__controls"
                >
                  <span
                    class="pf-v6-c-menu-toggle__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
          </div>
          <div
            class="pf-v6-c-toolbar__group pf-m-align-end"
          >
            <div
              class="pf-v6-c-toolbar__item"
            >
              <button
                aria-expanded="false"
                class="pf-v6-c-menu-toggle"
                data-ouia-component-id="OUIA-Generated-MenuToggle-5"
                data-ouia-component-type="PF6/MenuToggle"
                data-ouia-safe="true"
                type="button"
              >
                <span
                  class="pf-v6-c-menu-toggle__icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
                    />
                  </svg>
                </span>
                <span
                  class="pf-v6-c-menu-toggle__text"
                >
                  Last 1 min.
                </span>
                <span
                  class="pf-v6-c-menu-toggle__controls"
                >
                  <span
                    class="pf-v6-c-menu-toggle__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v6-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pf-v6-c-toolbar__content pf-m-hidden"
        hidden=""
      >
        <div
          class="pf-v6-c-toolbar__group"
        />
      </div>
    </div>
  </div>
  <div
    class="pf-v6-l-stack pf-m-gutter"
  >
    <div
      class="pf-v6-l-stack__item"
    >
      <div
        aria-label="Tcp traffic"
        class="pf-v6-c-card pf-m-expanded"
        data-ouia-component-id="OUIA-Generated-Card-1"
        data-ouia-component-type="PF6/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-v6-c-card__header"
        >
          <div
            class="pf-v6-c-card__header-toggle"
          >
            <button
              aria-disabled="false"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <span
                  class="pf-v6-c-card__header-toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pf-v6-c-card__header-main"
          >
            <div
              class="pf-v6-c-card__title"
            >
              <div
                class="pf-v6-c-card__title-text"
              >
                Tcp traffic
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-v6-c-card__expandable-content"
        >
          <div
            class="pf-v6-c-card__body"
            style="min-height: 450px;"
          >
            <div
              style="position: absolute; width: 100%; height: 100%; background-color: rgba(3, 3, 3, 0.1); left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 1;"
            >
              <div
                class="pf-v6-l-bullseye"
                data-testid="sk-loading-view"
              >
                <svg
                  aria-label="Contents"
                  aria-valuetext="Loading..."
                  class="pf-v6-c-spinner pf-m-xl"
                  role="progressbar"
                  style="--pf-v6-c-spinner--diameter: 150px;"
                  viewBox="0 0 100 100"
                >
                  <circle
                    class="pf-v6-c-spinner__path"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="45"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-l-stack__item"
    >
      <div
        aria-label="Tcp connections"
        class="pf-v6-c-card"
        data-ouia-component-id="OUIA-Generated-Card-2"
        data-ouia-component-type="PF6/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-v6-c-card__header"
        >
          <div
            class="pf-v6-c-card__header-toggle"
          >
            <button
              aria-disabled="false"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-2"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <span
                  class="pf-v6-c-card__header-toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pf-v6-c-card__header-main"
          >
            <div
              class="pf-v6-c-card__title"
            >
              <div
                class="pf-v6-c-card__title-text"
              >
                Tcp connections
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-l-stack__item"
    >
      <div
        aria-label="Tcp latency Out"
        class="pf-v6-c-card"
        data-ouia-component-id="OUIA-Generated-Card-3"
        data-ouia-component-type="PF6/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-v6-c-card__header"
        >
          <div
            class="pf-v6-c-card__header-toggle"
          >
            <button
              aria-disabled="false"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-3"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <span
                  class="pf-v6-c-card__header-toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pf-v6-c-card__header-main"
          >
            <div
              class="pf-v6-c-card__title"
            >
              <div
                class="pf-v6-c-card__title-text"
              >
                Tcp latency Out
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-l-stack__item"
    >
      <div
        aria-label="Tcp latency In"
        class="pf-v6-c-card"
        data-ouia-component-id="OUIA-Generated-Card-4"
        data-ouia-component-type="PF6/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-v6-c-card__header"
        >
          <div
            class="pf-v6-c-card__header-toggle"
          >
            <button
              aria-disabled="false"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-4"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <span
                  class="pf-v6-c-card__header-toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pf-v6-c-card__header-main"
          >
            <div
              class="pf-v6-c-card__title"
            >
              <div
                class="pf-v6-c-card__title-text"
              >
                Tcp latency In
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-l-stack__item"
    >
      <div
        aria-label="Http requests"
        class="pf-v6-c-card"
        data-ouia-component-id="OUIA-Generated-Card-5"
        data-ouia-component-type="PF6/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-v6-c-card__header"
        >
          <div
            class="pf-v6-c-card__header-toggle"
          >
            <button
              aria-disabled="false"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-5"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <span
                  class="pf-v6-c-card__header-toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pf-v6-c-card__header-main"
          >
            <div
              class="pf-v6-c-card__title"
            >
              <div
                class="pf-v6-c-card__title-text"
              >
                Http requests
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pf-v6-l-stack__item"
    >
      <div
        aria-label="Http responses"
        class="pf-v6-c-card"
        data-ouia-component-id="OUIA-Generated-Card-6"
        data-ouia-component-type="PF6/Card"
        data-ouia-safe="true"
        id=""
      >
        <div
          class="pf-v6-c-card__header"
        >
          <div
            class="pf-v6-c-card__header-toggle"
          >
            <button
              aria-disabled="false"
              class="pf-v6-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-6"
              data-ouia-component-type="PF6/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-v6-c-button__icon"
              >
                <span
                  class="pf-v6-c-card__header-toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v6-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pf-v6-c-card__header-main"
          >
            <div
              class="pf-v6-c-card__title"
            >
              <div
                class="pf-v6-c-card__title-text"
              >
                Http responses
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
